tests/unit/test_braille.py

```diff
@@ -0,0 +1,96 @@
+#tests/unit/test_braille.py
+#A part of NonVisual Desktop Access (NVDA)
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+#Copyright (C) 2017 NV Access Limited, Babbage B.V.
+
+"""Unit tests for the braille module.
+"""
+
+import unittest
+import braille
+from . import PlaceholderNVDAObject
+import controlTypes
+import api
+from config import conf
+import api
+import globalVars
+
+class NVDAObjectWithRole(PlaceholderNVDAObject):
+	"""An object that accepts a role as one of its construction parameters.
+	The name of the object will be set with the associated role label.
+	This class is used to quickly create objects for a fake focus ancestry."""
+
+	def __init__(self, role=controlTypes.ROLE_UNKNOWN,**kwargs):
+		super(NVDAObjectWithRole,self).__init__(**kwargs)
+		self.role=role
+
+	def _get_name(self):
+		return controlTypes.roleLabels.get(self.role,controlTypes.ROLE_UNKNOWN)
+
+class TestFocusContextPresentation(unittest.TestCase):
+	"""A test for the different focus context presentation options."""
+
+	@property
+	def regionsWithPositions(self):
+		return list(braille.handler.buffer.regionsWithPositions)
+
+	def setUp(self):
+		"""Set up a fake focus object and give it some ancestry."""
+		self.obj=NVDAObjectWithRole(role=controlTypes.ROLE_LISTITEM)
+		# Create and set a fake desktop object
+		self.fakeDesktopObj=NVDAObjectWithRole(role=controlTypes.ROLE_WINDOW)
+		api.setDesktopObject(self.fakeDesktopObj)
+		# Forcefully create a fake focus ancestry
+		globalVars.focusAncestors=[api.getDesktopObject(),NVDAObjectWithRole(role=controlTypes.ROLE_DIALOG),NVDAObjectWithRole(role=controlTypes.ROLE_LIST)]
+		braille.handler.handleGainFocus(self.obj)
+		# Make sure that we are testing with three regions
+		self.assertEqual(len(self.regionsWithPositions),3)
+
+	def test_fillDisplay(self):
+		"""Test for the case where both the focus object and its ancestors should be visible on a 40 cell display."""
+		conf['braille']['focusContextPresentation']="fill"
+		# Since we set the presentation mode, simulate another gainFocus so the regions will be updated properly
+		braille.handler.handleGainFocus(self.obj)
+		# WindowEndPos should be retrieved before we attempt to get the start position
+		# This is because getting windowEndPos can update windowStartPos
+		# Both the focus object and its parent should fit on the display
+		# Thus, the window end position is equal to the end position of the 3rd region
+		self.assertEqual(braille.handler.buffer.windowEndPos,self.regionsWithPositions[2][2])

```

Comment:
If it's not too much trouble, consider making the regions with positions code in braille return a namedtuple. That would make these tests more readable; e.g. `self.regionsWithPositions[2].end`.